internal/sequencers/espresso/input_box_reader.go

```diff
@@ -4,37 +4,59 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"log/slog"
 	"net/http"
 	"os"
+	"strconv"
+	"time"
 
 	"github.com/calindra/nonodo/internal/contracts"
 	"github.com/calindra/nonodo/internal/sequencers/inputter"
 	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/tidwall/gjson"
 )
 
+func getL1FinalizedTimestamp(espressoBlockHeight uint64) time.Time {
+	requestURL := fmt.Sprintf("https://query.cappuccino.testnet.espresso.network/availability/header/%d", espressoBlockHeight)
```

Comment:
It might be better to store this URL in a config var/const.

Reply:
refactored. 
I also notice the url appears here too https://github.com/Calindra/nonodo/blob/6df2655751cd8327956cf4571a650521627295fc/internal/dataavailability/espresso.go#L152
But should we pass the config all the way there too?  To do that, seems we need to change the `RollupAPI` or `Sequencer` here
https://github.com/Calindra/nonodo/blob/6df2655751cd8327956cf4571a650521627295fc/internal/rollup/rollup.go#L33
What do you think?